ts/Stock/Indicators/CMO/CMOIndicator.ts

```diff
@@ -0,0 +1,214 @@
+            y;
+        // calculate first point, check if the first value
+        // was added to sum of higher/lower values, and what was the value.
+
+        for (let j = period; j > 0; j--) {
+            if (values[j] > values[j - 1]) {
+                sumOfHigherValues += values[j] - values[j - 1];
+            } else if (values[j] < values[j - 1]) {
+                sumOfLowerValues += values[j - 1] - values[j];
+            }
+        }
+        // you might devide by 0 if all values are equal,
+        // so return 0 in this case.
+        y =
+            sumOfHigherValues + sumOfLowerValues > 0 ?
+                (100 * (sumOfHigherValues - sumOfLowerValues)) /
+                (sumOfHigherValues + sumOfLowerValues) :
+                0;
+
+        xData.push(xVal[period]);
+        yData.push(y);
+        CMO.push([xVal[period], y]);
+
+        for (i = period + 1; i < yValLen; i++) {
+            firstAddedSum = Math.abs(
+                values[i - period - 1] - values[i - period]
+            );
+            if (values[i] > values[i - 1]) {
+                sumOfHigherValues += values[i] - values[i - 1];
+            } else if (values[i] < values[i - 1]) {
+                sumOfLowerValues += values[i - 1] - values[i];
+            }
+
+            // Check, to which sum was the first value added to,
+            // and substract this value from given sum.
+            if (values[i - period] > values[i - period - 1]) {
+                sumOfHigherValues -= firstAddedSum;
+            } else {
+                sumOfLowerValues -= firstAddedSum;
+            }
+            // Same as above.
+            y =
+                sumOfHigherValues + sumOfLowerValues > 0 ?
+                    (100 * (sumOfHigherValues - sumOfLowerValues)) /
+                    (sumOfHigherValues + sumOfLowerValues) :
+                    0;
+
+            xData.push(xVal[i]);
+            yData.push(y);
+            CMO.push([xVal[i], y]);
+        }
+
+        return {
+            values: CMO,
+            xData: xData,
+            yData: yData
+        } as IndicatorValuesObject<TLinkedSeries>;
+    }
+}
+
+/* *
+ *
+ *  Prototype Properties
+ *
+ * */
+
+interface CMOIndicator {
+    pointClass: typeof CMOPoint;
+}
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+declare module '../../../Core/Series/SeriesType' {
+    interface SeriesTypeRegistry {
+        cmo: typeof CMOIndicator;
+    }
+}
+
+SeriesRegistry.registerSeriesType('cmo', CMOIndicator);
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default CMOIndicator;
+
+/**
+ * A `CMO` series. If the [type](#series.cmo.type) option is not
+ * specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.cmo
+ * @since     next
+ * @product   highstock
+ * @excluding dataParser, dataURL
+ * @requires  stock/indicators/indicators
+ * @requires  stock/indicators/cmo
+ * @apioption series.cmo
+ */
+
+(''); // to include the above in the js output
```

Comment:
Question to @bre1470 - does it matter if we end with `('')` or `''`?

Reply:
It's cute! Looks like it got compressed out, so it should be okay. 👍

Reply:
I agree! :D Looks like a happy robot sending you thumbs up!
`('');` ->   ('')👍